sct-commons/src/main/java/org/lfenergy/compas/sct/commons/scl/SclService.java

```diff
@@ -40,7 +44,15 @@ private SclService() {
 
     public static SclRootAdapter initScl(Optional<UUID> hId, String hVersion, String hRevision) throws ScdException {
         UUID headerId = hId.orElseGet(UUID::randomUUID);
-        return new SclRootAdapter(headerId.toString(), hVersion, hRevision);
+        SclRootAdapter scdAdapter = new SclRootAdapter(headerId.toString(), hVersion, hRevision);
+        TPrivate fileTypePrivate = new TPrivate();
+        fileTypePrivate.setType(CommonConstants.COMPAS_SCL_FILE_TYPE);
+        JAXBElement<TCompasSclFileType> compasFileType = new JAXBElement<>(
+                new QName(CompasExtensionsConstants.COMPAS_EXTENSION_NS_URI, CommonConstants.SCL_FILE_TYPE),
+                TCompasSclFileType.class, TCompasSclFileType.SCD);
+        fileTypePrivate.getContent().add(compasFileType);
+        scdAdapter.addPrivate(fileTypePrivate);
+        return scdAdapter;
```

Comment:
I think you should put this code in a new method and call it directly from the SclAutomationService, in order to divide the construction of SCD file with unitary steps